app/services/replacement_parameters_service.rb

```diff
@@ -45,9 +45,13 @@ def you_or_your
   # this should only be called when we aren't saving the output to our database
   # for example, when adding a link to send to the client only
   def sensitive_replacements
-    {
-      "Link.E-signature": client.login_link
-    }
+    if body.match(/<<\s*Link\.E-signature\s*>>/i)
+      {
+        "Link.E-signature": client.login_link
```

Comment:
This is kind of a "dangerous" method in that invoking it actually updates stuff on the client, but it kind of just looks like an attribute. We might consider changing this method name to "generate_login_link" to make it clear that its doing something other than just _providing_ the login link.